File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    let answer = [];
    let days = [];
    let front = 0;
    
    progresses.forEach((v, i) =>{
        days.push(Math.ceil((100 - v) / speeds[i]));
    })
    
    days.forEach((v, i) => {
        if(v > days[front]){
            answer.push(i - front);
            front = i;
        }
    })
    answer.push(days.length - front);
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from math import ceil

def solution(progresses, speeds):
    answer = []
    days = []
    front = 0

    # 소요시간을 정리해두자
    for i in range(len(progresses)):
        days.append(ceil((100 - progresses[i]) / speeds[i]))
    
    for idx in range(len(days)):
        if days[idx] > days[front]:
            answer.append(idx - front)
            front = idx
    answer.append(len(days) - front)

    return answer